docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,299 @@
+			```c
+			/****************************************************************************
+			 * Name: mpu_user_flash
+			 *
+			 * Description:
+			 *   Configure a region for user program flash
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size);
+			```
+			```c
+			/****************************************************************************
+			 * Name: mpu_userintsram
+			 *
+			 * Description:
+			 *   Configure a region as user internal SRAM
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size);
+			```
+		- For repeated configuration of a MPU region, a two-step process is required using below functions:
+			1. In the first step, we obtain the mpu register values and store them in an array using below function:
+			```c
+			/****************************************************************************
+			 * Name: mpu_get_register_value
+			 *
+			 * Description:
+			 *   Configure the user application SRAM mpu settings into the tcb variables
+			 *
+			 * Params:
+			 *  regs     : pointer to array in which to store the configured values
+			 *             If regs is NULL, then configure the mpu registers directly
+			 *  region   : number of the region to be configured
+			 *  base     : start address for the region
+			 *  size     : size of the region in bytes
+			 *  readonly : true indicates a readonly region
+			 *  execute  : true indicates that the region has execute permission
+			 ****************************************************************************/
+			 
+			void mpu_get_register_value(uint32_t *regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute);
+			```
+			2. In the second step, the values in the array are used to configure the mpu. This second step happens everytime during a context switch.
+			```c
+			/****************************************************************************
+			 * Name: up_set_mpu_app_configuration
+			 *
+			 * Description:
+			 *   Configure MPU for app using register values stored in rtcb's mpu_regs[]
+			 *
+			 * Params:
+			 *  rtcb : Common part of Task Control Block
+			 ****************************************************************************/
+			static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb);
+			```
+3. **Example : Adding new MPU region with RO setting**
+	1. Add your MPU region (say MPU_REG_NUM_XX) in enum MPU_REG_NUM in common mpu.h file.
```

Comment:
Could you give only real example without duplicated explanation like below ?
```
3. **Example : Adding new MPU region, MPU_REG_NUM_MY_REGION with RO setting**
    1. Add your MPU region in enum MPU_REG_NUM in common mpu.h file
    '''c
    enum MPU_REG_NUM {
     ....
     MPU_REG_NUM_MY_REGION,
     MPU_REG_NUM_MAX
    };
    '''

    2. To configure this MPU region as RO
    '''c
      mpu_ro(MPU_REG_NUM_MY_REGION, addr, RO_attr, ....);
    '''
```
I think we don't need to explain this again. It's almost the same as above 1st and 2nd.

Reply:
Added a real example as suggested.
(Example is inspired by how MPU is configured for Common Binary)

Reply:
Updated example with 2 steps.
Second step uses `up_mpu_set_register(...)` for setting mpu registers